CMakeLists.txt

```diff
@@ -0,0 +1,658 @@
+# Copyright (c) 2023-present The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://opensource.org/license/mit/.
+
+# Ubuntu 22.04 LTS Jammy Jellyfish, https://wiki.ubuntu.com/Releases, EOSS in June 2027:
+#  - CMake 3.22.1, https://packages.ubuntu.com/jammy/cmake
```

Comment:
if this refers to the docker base, we could update it to:
```suggestion
# Ubuntu 24.04 LTS Noble Numbat, https://wiki.ubuntu.com/Releases, EOSS in June 2029, EOL in April 2036:
#  - CMake 3.28.3, https://packages.ubuntu.com/noble/cmake
```

Reply:
> if this refers to the docker base...

It refers to a system that users are using to build Bitcoin Core. In most cases, it will be their regular machines. However, building Bitcoin Core within a Docker container is also supported.

Ubuntu and CentOS are mentioned here as mainstream DEB and RPM-based distros. However, the new CMake-based build system is designed to work out-of-the-box on a wide range of other platforms.

Reply:
Got it, so our tests are run with a different version than what we're mentioning here, right?
Not sure it makes sense, but could we have one of the docker images have this version, to expose ourselves to the documented os version as well (e.g. to catch if someone uses an old dependency, like a `CMake 3.22.1` incompatibility)?

Reply:
> Got it, so our tests are run with a different version than what we're mentioning here, right?

CMake version can be any >= 3.22. However, CMake ensures backward compatibility using the [policies](https://cmake.org/cmake/help/latest/manual/cmake-policies.7.html). Only policies known to version 3.22 are enabled, regardless of the actual CMake version.

> Not sure it makes sense, but could we have one of the docker images have this version, to expose ourselves to the documented os version as well (e.g. to catch if someone uses an old dependency, like a `CMake 3.22.1` incompatibility)?

That was done for the [staging branch](https://github.com/hebasto/bitcoin/tree/cmake-staging) (see `.github/workflows/cmake.yml`). For this PR, it is out of the scope as it is focused on the migration only. Anyway, [this](https://cirrus-ci.com/task/5479234460712960) CI job uses Ubuntu 22.04 and CMake 3.22.1.

Reply:
Thanks